vault/util_test.go

```diff
@@ -3,19 +3,40 @@
 
 package vault
 
-import "testing"
+import (
+	"math/rand"
+	"testing"
+)
+
+const n = 1024
 
 func TestMemZero(t *testing.T) {
-	b := []byte{1, 2, 3, 4}
+	r := rand.Intn(n)
+	b := randbytes(r)
 	memzero(b)
-	if b[0] != 0 || b[1] != 0 || b[2] != 0 || b[3] != 0 {
-		t.Fatalf("bad: %v", b)
+	if len(b) != r {
+		t.Fatalf("buffer has wrong length: %d", len(b))
+	}
+	for i := range b {
```

Comment:
```suggestion
	for i, v := range b {
```

Also works to get the value without having to reindex below.